src/tests/functional/inference_engine/transformations/smart_reshape/lstm_states_broadcast.cpp

```diff
@@ -0,0 +1,106 @@
+}
+
+
+class LSTMStatesBroadcastTestWithTI
+        : public testing::WithParamInterface<LSTMStatesAttributes>, public CommonTestUtils::TestsCommon {
+};
+
+TEST_P(LSTMStatesBroadcastTestWithTI, TI_With_LSTM) {
+    auto p = GetParam();
+
+    std::shared_ptr<ov::Model> model(nullptr);
+    {
+        auto X = std::make_shared<ov::opset9::Parameter>(p.data_et, ov::PartialShape{p.data_batch_size, 1,  p.input_size});
+        auto H_init = create_zero_constant(ov::element::i64, {1, p.hidden_size});
+        auto C_init = create_zero_constant(ov::element::i64, {1, p.hidden_size});
+
+        auto Xi = std::make_shared<ov::opset9::Parameter>(p.data_et, ov::PartialShape{1, 1, p.input_size});
+        auto H_t = std::make_shared<ov::opset9::Parameter>(p.data_et, ov::PartialShape{1, p.hidden_size});
+        auto C_t = std::make_shared<ov::opset9::Parameter>(p.data_et, ov::PartialShape{1, p.hidden_size});
+
+        // Body
+        auto squeeze = std::make_shared<ov::opset9::Squeeze>(Xi, create_constant<int64_t>({1}));
+        auto W = create_zero_constant(p.data_et, {p.hidden_size * 4, p.input_size});
+        auto R = create_zero_constant(p.data_et, {p.hidden_size * 4, p.hidden_size});
+
+        auto lstm_cell = std::make_shared<ov::opset9::LSTMCell>(squeeze, H_t, C_t, W, R, static_cast<size_t>(p.hidden_size.get_length()));
+        auto res_1 = std::make_shared<ov::opset9::Result>(lstm_cell->output(0));
+        auto unsqueeze = std::make_shared<ov::opset9::Unsqueeze>(lstm_cell->output(0), create_constant<int64_t>({1}));
+        auto res_2 = std::make_shared<ov::opset9::Result>(unsqueeze);
+        auto res_3 = std::make_shared<ov::opset9::Result>(lstm_cell->output(1));
+        auto body = std::make_shared<ov::Model>(ov::OutputVector{res_1, res_2, res_3}, ov::ParameterVector{Xi, H_t, C_t});
+
+        auto tensor_iterator = std::make_shared<ov::opset9::TensorIterator>();
+        tensor_iterator->set_body(body);
+
+        tensor_iterator->set_merged_input(C_t, C_init, res_3);
+        tensor_iterator->set_sliced_input(Xi, X, 0, 1, 1, -1, 1);
+        tensor_iterator->set_merged_input(H_t, H_init, res_1);
+
+        auto out0 = tensor_iterator->get_iter_value(res_1, -1);
+        auto out1 = tensor_iterator->get_concatenated_slices(res_2, 0, 1, 1, -1, 0);
+
+        auto res_ti_1 = std::make_shared<ov::opset9::Result>(tensor_iterator->output(1));
+        auto res_ti_2 = std::make_shared<ov::opset9::Result>(tensor_iterator->output(0));
+        model = std::make_shared<ov::Model>(ov::NodeVector{res_ti_1, res_ti_2},
+                                               ov::ParameterVector{X});
+    }
+    model->reshape(ov::PartialShape{p.new_batch_size, 1, p.input_size});
+    ASSERT_NO_THROW(model->reshape(ov::PartialShape{p.new_batch_size, 1, p.input_size}));
+}
+
+static std::vector<LSTMStatesAttributes> params = {
+        LSTMStatesAttributes{ov::element::f32, {1}, {2}, {512}, {256}},
```

Comment:
It looks like only 2nd parameter is changing. 
Will be there more test cases added type, size etc?

if not maybe move not changing values as test const parameter and use fixture with only one parameter.

Reply:
I'd like to keep this test configurable in case we would have some bugs in the code and will need to simulate that in test.